state/redis/redis_query.go

```diff
@@ -295,10 +295,49 @@ func (q *Query) execute(ctx context.Context, client rediscomponent.RedisClient)
 	if err != nil {
 		return nil, "", err
 	}
-	arr, ok := ret.([]interface{})
+
+	res := []state.QueryItem{}
+
+	arr, ok := ret.([]any)
```

Comment:
Can you extract the pre- and post- in 2 separate functions please, to keep the code more readable

Reply:
Good idea, done :slightly_smiling_face: